src/auth/auth.controller.ts

```diff
@@ -89,4 +90,29 @@ export class AuthController {
       accessToken: await this.authService.issueAccessToken(token),
     }
   }
+
+  private createOauthState(returnUrl: string, overrideBackendUrl?: string) {
+    const state: OauthStatePayload = {
+      returnUrl: returnUrl || this.configService.get('backendPublicUrl'),
+    }
+    if (overrideBackendUrl) {
+      state.overrideBackendUrl = overrideBackendUrl
+    }
```

Comment:
why not this?
```suggestion
    const state: OauthStatePayload = {
      returnUrl: returnUrl || this.configService.get('backendPublicUrl'),
      overrideBackendUrl: overrideBackendUrl
    }
```

Reply:
Don't want that key in the object when it's undefined

Reply:
if it's undefined, it would be undefined, no?

Reply:
Hmm actually `undefined` is not a valid type in JSON, so `JSON.parse` will remove it